crates/turbopack-css/src/references/import.rs

```diff
@@ -344,13 +218,40 @@ impl ImportAssetReference {
 #[turbo_tasks::value_impl]
 impl ModuleReference for ImportAssetReference {
     #[turbo_tasks::function]
-    fn resolve_reference(&self) -> Vc<ModuleResolveResult> {
-        css_resolve(
+    async fn resolve_reference(&self) -> Result<Vc<ModuleResolveResult>> {
+        let css_asset = Vc::try_resolve_downcast_type::<CssModuleAsset>(self.origin).await?;
```

Comment:
I think I would prefer to just pass the previous `ImportContext` as additional argument to `ImportAssetReference::new`